pkg/volume/gce_pd/gce_pd.go

```diff
@@ -234,16 +237,31 @@ func (plugin *gcePersistentDiskPlugin) NewDeleter(spec *volume.Spec) (volume.Del
 }
 
 func (plugin *gcePersistentDiskPlugin) newDeleterInternal(spec *volume.Spec, manager pdManager) (volume.Deleter, error) {
-	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.GCEPersistentDisk == nil {
-		return nil, fmt.Errorf("spec.PersistentVolumeSource.GCEPersistentDisk is nil")
+	zones := make(sets.String)
+	var err error
+	if spec.PersistentVolume != nil {
+		if spec.PersistentVolume.Spec.GCEPersistentDisk == nil {
+			return nil, fmt.Errorf("spec.PersistentVolumeSource.GCEPersistentDisk is nil")
+		}
+
+		zoneLabel, ok := spec.PersistentVolume.Labels[apis.LabelZoneFailureDomain]
+		if ok && zoneLabel != "" {
+			zones, err = util.LabelZonesToSet(zoneLabel)
+			if err != nil {
+				return nil, err
+			}
+		}
 	}
+
 	return &gcePersistentDiskDeleter{
 		gcePersistentDisk: &gcePersistentDisk{
 			volName: spec.Name(),
 			pdName:  spec.PersistentVolume.Spec.GCEPersistentDisk.PDName,
 			manager: manager,
 			plugin:  plugin,
-		}}, nil
+		},
+		zones: zones,
```

Comment:
As mentioned above see if we can just pass bool here, and reuse the same code as other calls. Other wise remember to update the code that pulls zone set for https://github.com/kubernetes/kubernetes/pull/67530 here as well.

Reply:
`DeleteDisk()` takes the zone as a parameter. The gcepd Deleter object itself doesn't have access to the spec directly; the necessary information (including zone information) has to be pulled off the spec at Deleter construction